packages/react-ui/src/app/builder/step-settings/piece-settings/connection-select.tsx

```diff
@@ -40,41 +47,87 @@ const ConnectionSelect = React.memo((params: ConnectionSelectProps) => {
     pieceName: params.piece.name,
     cursor: undefined,
     limit: 100,
-    projectId: authenticationSession.getProjectId(),
+    projectId: authenticationSession.getProjectId() ?? '',
   });
-
-  const addBrackets = (str: string) => `{{connections['${str}']}}`;
+  const currentlySelectedConnection = useMemo(() => {
+    const connectionName = removeBrackets(
+      form.getValues().settings.input.auth ?? '',
+    );
+    return (
+      connectionsPage?.data?.find(
+        (connection) => connection.name === connectionName,
+      ) ?? null
+    );
+  }, [connectionsPage?.data, form.getValues().settings.input.auth]);
 
   return (
     <FormField
       control={form.control}
       name={'settings.input.auth'}
       render={({ field }) => (
-        <FormItem>
-          <CreateOrEditConnectionDialog
-            piece={params.piece}
-            onConnectionCreated={(connectionName) => {
-              refetch();
-              field.onChange(addBrackets(connectionName));
-            }}
-            open={connectionDialogOpen}
-            setOpen={setConnectionDialogOpen}
-          ></CreateOrEditConnectionDialog>
-          <FormLabel>{t('Connection')}</FormLabel>
-          <Select
-            open={selectConnectionOpen}
-            onOpenChange={setSelectConnectionOpen}
-            defaultValue={field.value as string | undefined}
-            onValueChange={field.onChange}
-            disabled={params.disabled}
-          >
-            <SelectTrigger>
-              <SelectValue placeholder={t('Select a connection')} />
-            </SelectTrigger>
-            <SelectContent>
-              {isLoading && <SelectLoader />}
-              {!isLoading && (
-                <>
+        <>
+          {isLoading && (
+            <Select disabled={params.disabled}>
+              <SelectContent>
+                <SelectLoader />
+              </SelectContent>
+            </Select>
+          )}
+          {!isLoading && (
+            <FormItem>
+              <CreateOrEditConnectionDialog
+                reconnectConnection={currentlySelectedConnection}
```

Comment:
This breaks +New Connection functionality

Reply:
fixed.